src/main/java/games/strategy/engine/lobby/client/login/LobbyLogin.java

```diff
@@ -35,85 +37,72 @@ public LobbyLogin(final Window parent, final LobbyServerProperties lobbyServerPr
    * If we could not login, return null.
    * </p>
    */
-  public LobbyClient login() {
+  public @Nullable LobbyClient login() {
     if (!lobbyServerProperties.isServerAvailable()) {
-      JOptionPane.showMessageDialog(
-          parentWindow,
-          lobbyServerProperties.serverErrorMessage,
-          "Could not connect to server",
-          JOptionPane.ERROR_MESSAGE);
+      showError("Could not connect to server", lobbyServerProperties.serverErrorMessage);
       return null;
     }
     if (lobbyServerProperties.port == -1) {
-      JOptionPane.showMessageDialog(parentWindow,
+      showError("Could not connect to server",
           "<html>Could not find lobby server for this version of TripleA, <br>"
               + "Please make sure you are using the latest version: " + UrlConstants.LATEST_GAME_DOWNLOAD_WEBSITE
-              + "</html>",
-          "Could not connect to server", JOptionPane.ERROR_MESSAGE);
+              + "</html>");
       return null;
     }
     return loginToServer();
   }
 
-  private LobbyClient login(final LoginPanel panel) {
+  private @Nullable LobbyClient login(final LoginPanel panel) {
     try {
-      final String mac = MacFinder.getHashedMacAddress();
-      final ClientMessenger messenger = new ClientMessenger(lobbyServerProperties.host, lobbyServerProperties.port,
-          panel.getUserName(), mac, new IConnectionLogin() {
-            private final AtomicReference<String> internalError = new AtomicReference<>();
-
-            @Override
-            public void notifyFailedLogin(String message) {
-              if (internalError.get() != null) {
-                message = internalError.get();
-              }
-              JOptionPane.showMessageDialog(parentWindow, message, "Login Failed", JOptionPane.ERROR_MESSAGE);
-            }
-
-            @Override
-            public Map<String, String> getProperties(final Map<String, String> challengeProperties) {
-              final Map<String, String> props = new HashMap<>();
-              if (panel.isAnonymousLogin()) {
-                props.put(LobbyLoginValidator.ANONYMOUS_LOGIN, Boolean.TRUE.toString());
-              } else {
-                final boolean isUpdatedLobby = RsaAuthenticator.canProcessChallenge(challengeProperties);
-                String salt = challengeProperties.get(LobbyLoginValidator.SALT_KEY);
-                if (salt == null) {
-                  if (!isUpdatedLobby) {
-                    // the server does not have a salt value
-                    // so there is no user with our name,
-                    // continue as before
-                    internalError.set("No account with that name exists");
-                  }
```

Comment:
This special case was removed in the updated code below.

Primarily, we are currently running an "updated" lobby, so this code should never be hit.  Secondarily, it would have been a pain trying to communicate this custom error message back to the caller after I removed `IConnectionLogin#notifyFailedLogin()`. :smile:

Reply:
just wanna say good work and appreciate the time you put in